app/src/components/StyledButton/StyledButton.module.css

```diff
@@ -0,0 +1,267 @@
+.typeA {
+  --line_color: #555555;
+  --back_color: #ffecf6;
+}
+.typeB {
+  --line_color: #1b1919;
+  --back_color: #e9ecff;
+}
+.typeC {
+  --line_color: #000000;
+  --back_color: #defffa;
+}
+.button {
+  position: relative;
+  z-index: 0;
+  width: 240px;
+  height: 56px;
+  text-decoration: none;
+  font-size: 14px;
+  font-weight: bold;
+  color: var(--line_color);
+  letter-spacing: 2px;
+  transition: all 0.3s ease;
+  background-color: inherit;
+}
+.buttonText {
+  display: flex;
+  font-size: 13px;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+}
+.button::before,
+.button::after,
+.buttonText::before,
+.buttonText::after {
+  content: "";
+  position: absolute;
+  height: 3px;
+  border-radius: 2px;
+  background: var(--line_color);
+  transition: all 0.5s ease;
+}
+.button::before {
+  top: 0;
+  left: 54px;
+  width: calc(100% - 56px * 2 - 16px);
+}
+.button::after {
+  top: 0;
+  right: 54px;
+  width: 8px;
+}
+.buttonText::before {
+  bottom: 0;
+  right: 54px;
+  width: calc(100% - 56px * 2 - 16px);
+}
+.buttonText::after {
+  bottom: 0;
+  left: 54px;
+  width: 8px;
+}
+.buttonLine {
+  position: absolute;
+  top: 0;
+  width: 56px;
+  height: 100%;
+  overflow: hidden;
+}
+.buttonLine::before {
+  content: "";
+  position: absolute;
+  top: 0;
+  width: 150%;
+  height: 100%;
+  box-sizing: border-box;
+  border-radius: 300px;
+  border: solid 3px var(--line_color);
+}
+.buttonLine:nth-child(1),
+.buttonLine:nth-child(1)::before {
+  left: 0;
+}
+.buttonLine:nth-child(2),
+.buttonLine:nth-child(2)::before {
+  right: 0;
+}
+.button:hover {
+  letter-spacing: 6px;
+}
+.button:hover::before,
+.button:hover .buttonText::before {
+  width: 8px;
+}
+.button:hover::after,
+.button:hover .buttonText::after {
+  width: calc(100% - 56px * 2 - 16px);
+}
+.buttonDrow1,
+.buttonDrow2 {
+  position: absolute;
+  z-index: -1;
+  border-radius: 16px;
+  transform-origin: 16px 16px;
+}
+.buttonDrow1 {
```

Comment:
similar comment, maybe a better name here? Think about it like you having to work with another developer. If the name was descriptive, they wouldn't have to either go through the rest the code or bug you :)